src/Units/MoveTypes/MoveTypeFey.java

```diff
@@ -0,0 +1,53 @@
+package Units.MoveTypes;
+
+import Terrain.Environment;
+import Units.Unit;
+import Engine.FloodFillFunctor;
+import Engine.XYCoord;
+import Engine.FloodFillFunctor.BasicMoveFillFunctor;
+import Terrain.GameMap;
+
+/**
+ * A variant MoveType that is barred from entering non-friendly ownable healing terrain
+ */
+public class MoveTypeFey extends MoveType
```

Comment:
Interesting name - why?

Reply:
I was initially planning to make just `DoRFloat` but that's super lame and uninformative.
This is generic *and* has a really evocative/informative name.

Reply:
Evocative, yes. Informative I might dispute.

Reply:
Everybody knows the Fey can't enter your house without your permission.
It's like, a staple of western folklore.
In any case, I *also* put that meaning in the javadoc.